### ISSUE TITLE
Make default for fields @PolyDet

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#115]
Currently, the default for fields is `@Det`. I believe `@PolyDet` would be best. Firstly, the default for method and constructor parameters is `@PolyDet`, meaning the following code won't type check by default:

```java
import org.checkerframework.checker.determinism.qual.*;

public class TestDefaults {
    int field;

    public TestDefaults(@PolyDet int field) {
        this.field = field;
    }
}
```
```
TestDefaults.java:8: error: [invalid.field.assignment] cannot assign into @Det field via @PolyDet expression
        this.field = field;
            ^
```

Since the default for constructor parameters is `@PolyDet`, this kind of error would come up in almost any constructor by default.

Secondly, intuitively a field should depend on the determinism of the enclosing class. So, if the enclosing class is not declared `@Det`, the most logical choice would be for the fields to be `@PolyDet` themselves.

**Comment by smillst** [comment]
1. `@PolyDet` should only be the default for instance fields.
2.  This also means that every class with an instance field will need to be annotated as `@HasQualifierParameter`.

**Comment by t-rasmud** [comment]
Fixed in cd26ae0
